src/scripts/yinglation.js

```diff
@@ -279,6 +536,7 @@ const caseMatch = (str0, str1) => {
 };
 const replacer = word => {
   if (word.toLowerCase() in map) return caseMatch(word, map[word.toLowerCase()]);
+  else if (word.toLowerCase() in zMap) return caseMatch(word, zMap[word.toLowerCase()]);
```

Comment:
should be `else if (!reverseMode && word.toLowerCase() in zMap)` so that it doesn't still translate words with the `z` phoneme when in reverse mode